packages/strapi/lib/middlewares/logger/index.js

```diff
@@ -25,14 +26,31 @@ module.exports = strapi => {
     initialize() {
       const { level, exposeInContext, requests } = strapi.config.middleware.settings.logger;
 
-      if (level) {
-        strapi.log.level = level;
+      const logLevels = ['fatal', 'error', 'warn', 'info', 'debug', 'trace'];
+
+      if (!_.includes(logLevels, level)) {
+        throw new Error(
+          "Invalid log level set in middleware configuration. Accepted values are: '" +
+            logLevels.join("', '") +
+            "'."
+        );
       }
 
+      strapi.log.level = level;
```

Comment:
I think instead of lodding an error we can overwritte the log.level in the file with the env var if it is set. that would make it more consistent don't you think ?

Reply:
If that's ok. It would make sense, but should it be considered as a breaking change?

Reply:
good point let's keep it this way for now